### ISSUE TITLE
Remove name field from keras application models

### ISSUE THREAD

**Issue by Mainak431** [keras-team/keras#12584]
Keras application models had name field in every element. Like - 'mixed0', 'mixed1', 'mixed2' ,'avgpool' etc. Please remove the fields as they are very contradicting when more than one such application  is used in the single model. 

Like if i do - 

base_model1 = keras.applications.inception_v3.InceptionV3(weights=None, input_shape=(img_rows, img_cols, img_channels),
                                                         include_top=True, pooling='max')

base_model2 = keras.applications.inception_v3.InceptionV3(weights=None, input_shape=(img_rows, img_cols, img_channels),
                                                         include_top=True, pooling='max')

x = base_model1.output
x2 = base_model2.output

x = keras.layers.concatenate([x,x2], axis = 1)

And then if try to create model. Then it will give error like - 'mixed0' name used twice. 

This name field is restricting the use the this readymade models. 
Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
